ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -0,0 +1,123 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <onnx/onnx_pb.h>
+
+#include "onnx_import/editor/editor.hpp"
+#include "onnx_import/utils/parser.hpp"
+
+using namespace ngraph;
+
+namespace
+{
+    using namespace ONNX_NAMESPACE;
+
+    const std::map<element::Type_t, TensorProto_DataType> NG_2_ONNX_TYPES = {
+        {element::Type_t::bf16, TensorProto_DataType::TensorProto_DataType_BFLOAT16},
+        {element::Type_t::f16, TensorProto_DataType::TensorProto_DataType_FLOAT16},
+        {element::Type_t::f32, TensorProto_DataType::TensorProto_DataType_FLOAT},
+        {element::Type_t::f64, TensorProto_DataType::TensorProto_DataType_DOUBLE},
+        {element::Type_t::i8, TensorProto_DataType::TensorProto_DataType_INT8},
+        {element::Type_t::i16, TensorProto_DataType::TensorProto_DataType_INT16},
+        {element::Type_t::i32, TensorProto_DataType::TensorProto_DataType_INT32},
+        {element::Type_t::i64, TensorProto_DataType::TensorProto_DataType_INT64},
+        {element::Type_t::u8, TensorProto_DataType::TensorProto_DataType_UINT8},
+        {element::Type_t::u16, TensorProto_DataType::TensorProto_DataType_UINT16},
+        {element::Type_t::u32, TensorProto_DataType::TensorProto_DataType_UINT32},
+        {element::Type_t::u64, TensorProto_DataType::TensorProto_DataType_UINT64},
+    };
+
+    ValueInfoProto* find_graph_input(GraphProto& graph, const std::string& name)
+    {
+        for (int i = 0; i < graph.input_size(); ++i)
+        {
+            auto* input_desc = graph.mutable_input(i);
+            if (input_desc->has_name() && input_desc->name() == name)
+            {
+                return input_desc;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void modify_input_type(ValueInfoProto& onnx_input, element::Type_t elem_type)
+    {
+        const std::string malformed_input_error_msg =
+            "The input is malformed, cannot change the data type. Input name: " + onnx_input.name();
+
+        if (!onnx_input.has_type())
+        {
+            throw ngraph_error(malformed_input_error_msg);
```

Comment:
I think on the negative paths, we can have a specific error message (e.g. here "Input is malformed - it has no type" and similar in line 71) so the error message is not created every time  modify_input_type is called, but only on the negative path (which should be rare).